qiskit_dynamics/solvers/solver_functions.py

```diff
@@ -174,14 +196,22 @@ def solve_lmde(
     Optional arguments for any of the solver routines can be passed via ``kwargs``.
     Available LMDE-specific methods are:
 
-    - ``'scipy_expm'``: A matrix-exponential solver using ``scipy.linalg.expm``.
-      Requires additional kwarg ``max_dt``, indicating the maximum step
+    - ``'scipy_expm'``: A fixed-step matrix-exponential solver using ``scipy.linalg.expm``.
+      Requires additional kwarg ``max_dt`` indicating the maximum step
       size to take. This solver will break integration periods into even
-      sub-intervals no larger than ``max_dt``, and solve over each sub-intervals via
+      sub-intervals no larger than ``max_dt``, and solve over each sub-interval via
       matrix exponentiation of the generator sampled at the midpoint.
     - ``'jax_expm'``: JAX-implemented version of ``'scipy_expm'``, with the same arguments and
       logic.
-
+    - ``'jax_expm_parallel'``: Same as ``'jax_expm'``, however all loops are implemented using
```

Comment:
Is there any reason to have a separate method for these, rather than just making "parallel" a kwarg option for these jax methods that you can set to True or False?

Reply:
I think this is a good idea. One slightly awkward thing though is that `jax_RK4_parallel` is an LMDE method, whereas `jax_RK4` is an ODE method, in the sense that, even though RK4 is an ODE method, the approach to parallelization is based on LMDE structure.

This will require a change in how `solve_lmde` delegates to `solve_ode`; right now it assumes that the list of methods available in each has no overlap, and now we'll need to do some checking based on additional kwargs. I think this is fine, but will just increase the complexity of that step a little bit.

Reply:
I'm now of the mind that this should remain as is - just tried changing it and due to RK4 being an ODE method, but the parallel version is an LMDE method, it makes the logic to check when `solve_lmde` delegates to `solve_ode` very weird. (E.g. we will need to add `jax_RK4` to the list of both `solve_ode` and `solve_lmde` methods.) The current way feels a little less elegant but a lot simpler/obvious.